packages/util/src/unique-name-generator.ts

```diff
@@ -0,0 +1,214 @@
+/**
+ * Holds state on already used names and provides new names if there are naming conflicts.
+ */
+export class UniqueNameGenerator {
```

Comment:
What was the reason you went from `finder` to `generator`? I have no strong opinion just want to now the reason.

Reply:
I liked the notion of a generator more, because `finder` implies that it already exists and only has to be filtered from something (like Array.find).
What we do is "get" the index based on some logic and "create" a new name using this. I then thought of the concept of generators in Python and decided that this is close enough.
As this is now public API, I just found it a bit more on the point.

I am open to discuss this, though. Changing it back or to something else isn't a big deal.